atom/browser/ui/views/menu_bar.cc

```diff
@@ -24,26 +24,62 @@ namespace atom {
 
 namespace {
 
-const char kViewClassName[] = "ElectronMenuBar";
+#if defined(USE_X11)
+#if GTK_CHECK_VERSION(3, 0, 0)
 
-// Default color of the menu bar.
-const SkColor kDefaultColor = SkColorSetARGB(255, 233, 233, 233);
+SkColor GdkRgbaToSkColor(const GdkRGBA& rgba) {
+  return SkColorSetARGB(rgba.alpha*255, rgba.red*255,
+                        rgba.green*255, rgba.blue*255);
+}
 
-#if defined(USE_X11)
+SkColor GetStyleContextFgColor(GtkStyleContext* style_context,
+                               GtkStateFlags state) {
+  GdkRGBA rgba;
+  gtk_style_context_get_color(style_context, state, &rgba);
+  return GdkRgbaToSkColor(rgba);
+}
+
+SkColor GetStyleContextBgColor(GtkStyleContext* style_context,
+                               GtkStateFlags state) {
+  GdkRGBA rgba;
+  gtk_style_context_get_background_color(style_context, state, &rgba);
+  return GdkRgbaToSkColor(rgba);
+}
+
+// GTK+ 3 impl
 void GetMenuBarColor(SkColor* enabled, SkColor* disabled, SkColor* highlight,
                      SkColor* hover, SkColor* background) {
   GtkWidget* menu_bar = gtk_menu_bar_new();
+  GtkStyleContext* sc = gtk_widget_get_style_context(menu_bar);
+  *enabled    = GetStyleContextFgColor(sc, GTK_STATE_FLAG_NORMAL);
+  *disabled   = GetStyleContextFgColor(sc, GTK_STATE_FLAG_INSENSITIVE);
+  *highlight  = GetStyleContextFgColor(sc, GTK_STATE_FLAG_SELECTED);
+  *hover      = GetStyleContextFgColor(sc, GTK_STATE_FLAG_PRELIGHT);
+  *background = GetStyleContextBgColor(sc, GTK_STATE_FLAG_NORMAL);
+  g_object_unref(G_OBJECT(menu_bar));
+}
 
+#else  // GTK_CHECK_VERSION(3,0,0)
+
+// GTK+ 2 impl
```

Comment:
Do we still need GTK 2 implementation?

Reply:
Good question! I'll be back in the office on Thursday and will revisit this patch then and find out about GTK 2

Reply:
As one would expect, `electron` is linked against `gtk3`:

```sh
$ ldd ./electron | grep gtk
libgtk3ui.so => /home/charles/src/electron/electron/out/D/./libgtk3ui.so (0x00007ff3cd4d5000)
libgtk-3.so.0 => /usr/lib/x86_64-linux-gnu/libgtk-3.so.0 (0x00007ff3c9986000)
```

and I don't see any conditionals in `electron/electron` or `electron/libchromiumcontent` that would let a build choose between 2 and 3... so I suspect it's safe & desirable to remove the references to 2 and I'll do that in a followup push today